sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -872,6 +886,25 @@ object PushDownPredicate extends Rule[LogicalPlan] with PredicateHelper {
       pushDownPredicate(filter, u.child) { predicate =>
         u.withNewChildren(Seq(Filter(predicate, u.child)))
       }
+
+    case filter @ Filter(condition, watermark: EventTimeWatermark) =>
```

Comment:
Why not changing `EventTimeWatermark ` to `UnaryNode`? Then, we do not need to write a separate case only for `EventTimeWatermark`. We can reuse the existing `pushDownPredicate`, right?

We also have the other rules that already consider `UnaryNode`, do you think it make sense to avoid duplicating the codes for `EventTimeWatermark ` only?